File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        sort(arr.begin(),arr.end());
        vector<bool> freq(1001,false);
        int cnt=1, prev=arr[0];
        for(int i=1;i<(int)arr.size();++i){
            if(arr[i]!=prev){
                if(freq[cnt]) return false;
                freq[cnt]=true;
                cnt=1;
                prev=arr[i];
            }
            else{
                ++cnt;
            }
        }
        if(freq[cnt]) return false;
        return true;
    }
};

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        int n= arr.length, count;
        HashMap<Integer,Integer> freqOfElem = new HashMap<>();
        for(int i=0;i<n;i++){
            count= 0;
            if(freqOfElem.containsKey(arr[i])){
                count= freqOfElem.get(arr[i]);
            }
            freqOfElem.put(arr[i],count+1);
        }
        HashSet<Integer> hs = new HashSet<>();
        for(int k: freqOfElem.keySet()){
            if(hs.contains(freqOfElem.get(k))){
                return false;
            }
            hs.add(freqOfElem.get(k));
        }
        return true;
    }
}